src/controller/user.controller.ts

```diff
@@ -0,0 +1,19 @@
+import type { UserService } from '../services/user.service.js';
+import  type { CreateUserInput } from '../types/types.js';
+
+export class UserController {
+  private userService: UserService;
+
+  constructor(userService: UserService) {
+    this.userService = userService;
+  }
+
+  async createUser(params: CreateUserInput) {
+    try {
+      return await this.userService.createUser(params);
```

Comment:
Não precisamos do `await` quando damos um `return` 
```suggestion
      return this.userService.createUser(params);
```

Reply:
Beleza!, isso porque  como não to precisando manipular o valor retornado ou lidar com erros antes de retornar, o await é desnecessário no return né?, além do fato que no resolver, já esta sendo resolvido a promise